src/types/mod.rs

```diff
@@ -113,48 +89,52 @@ impl Symbol for Op {
 #[derive(Debug, Clone, Eq)]
 pub struct Var {
     id: DeBruijn,
-    name: Name_,
+    name: Option<String>,
 }
 impl Var {
-    pub fn new(id: DeBruijn, name: Name_) -> Self {
+    pub fn new(id: DeBruijn, name: Option<String>) -> Self {
         Var { id, name }
     }
+    pub fn named(mut self, name: String) -> Self {
+        self.name = Some(name);
+        self
+    }
```

Comment:
Hmm... doesn't this potentially introduce legibility problems if we allow people to change the names of variables after they've been created (e.g. `S x_ y_ z_` could turn into `S x_ x_ y_`, which looks very different from what the underlying identitites would show).

Reply:
This will no longer be necessary if we switch to `new_distinct` as I mentioned above, which takes an optional name.